docs/js/emitter.md

```diff
@@ -8,27 +8,43 @@ path: emitter
 ---
 
 # Emitter
-`Emitter` is an instance of [node's eventEmitter](https://nodejs.org/api/events.html), responsible to handle the communication between garden components
+An emitter is an object responsible for handling the communication among components.
 
-Usually garden components will emit an event to allow third party integrations with it.
+This element is built as an instance of the [EventEmitter](https://nodejs.org/api/events.html) object from Node.js and therefore exposes some useful functions, such as `emitter.on()` and `emitter.emit()`.
 
-For example, the [validation](validation.html) component emits an event when a field is validated. Either for success or error:
 
+### Registering events
+To register an event you can use the `emitter.on()` function, passing the event name and a handler as parameters.
 ```js
 import emitter from 'src/utils/emitter';
 
+emitter.on('eventName', () => {
+  // Here you can handle the event.
+});
+```
+### Passing arguments
+It is possible to pass arguments to a listener while emitting an event with the `emitter.emit()` function.
```

Comment:
I personally don't like using the contract form in cases like this. It sounds clearer to me this way, but let's see what others have to say :)